src/Digdir.Domain.Dialogporten.WebApi/Common/Extensions/ExceptionHandlerExtensions.cs

```diff
@@ -22,7 +22,9 @@ public static IApplicationBuilder UseProblemDetailsExceptionHandler(this IApplic
                 var error = exHandlerFeature.Error.Message;
                 var logger = ctx.Resolve<ILogger<ExceptionHandler>>();
                 logger.LogError(exHandlerFeature.Error, "{@Http}{@Type}{@Reason}", http, type, error);
-                ctx.Response.StatusCode = StatusCodes.Status500InternalServerError;
+                ctx.Response.StatusCode = type == nameof(HttpRequestException)
```

Comment:
Both upstream network errors and `HttpResponseMessage.EnsureSuccessStatusCode()` will throw `HttpRequestException`, but this is a bit hacky. Other exceptions might be `SyntheticTimeoutException` from FusionCache if factory timeouts are reached, or `JsonException` if the response is borked somehow (but still has 2xx code). Is there any other way we can somehow flag within the context that a upstream request is being attempted, and clear it after successful response/deserialization? So regardless of exception thrown, if that flag is set, we return `502 Bad Gateway`?

Reply:
* Wrapper exception with flag? 
* A list/map of exceptions in this ExceptionHandler that maps ExeptionType to StatusCode?

Reply:
I went the with the wrapper exception.